src/core/mediator/UnitsHelper.cpp

```diff
@@ -0,0 +1,38 @@
+#include "UnitsHelper.hpp"
+
+ut_system* UnitsHelper::unit_system;
+std::once_flag UnitsHelper::unit_system_inited;
+
+double UnitsHelper::get_converted_value(const std::string &in_units, double value, const std::string &out_units)
+{
+    if(in_units == out_units){
+        return value; // Early-out optimization
+    }
+    if(in_units == "" || out_units == ""){
+        return value; //TODO: Warn? Make behavior configurable?
```

Comment:
I don't think we want to do this.  If nothing else, this isn't going to throw a _runtime_error_ that will trigger the warning message.

Further, aside from being another place to possibly forget to go back to if we add configurability, what scenario is there that we need to support something not providing ~~valid~~ any units for itself?  Sure, it makes sense to just do value pass-thru for now if there is a hangup in _**the framework**_ recognizing how to convert between models' units.  But I'm pretty sure this case would only occur when a model (or a forcing data provider source) didn't provide units at all; i.e., a hangup in something external to the framework.  That seems different enough that I think we need to throw an exception that won't be caught and trigger the pass-thru behavior, even at this point.   Not sure whether that means using something other than _runtime_error_ here or changing the places upstream that call this and handle catching that type.

This one is complex, so let me know if you want to discuss further.

Reply:
Good question. I think I hit this when testing and one of the C++ models did this... or the legacy `Forcing` object... really I think this block existed primarily as an optimization: if it isn't there, then the unit strings just get passed to `ut_parse` and possibly `ut_get_converter` and somewhere along the line it will fail to convert... and I don't know what the execution costs are on those functions and they will be run a _lot_. Basically we probably need to do here whatever we would do in a unit conversion failure (in this case one or the other is _not_ blank, otherwise the previous block would have returned) but leave it here for performance reasons (unless we forcibly abort whenever a unit conversion fails).

I think I had another scenario in mind where a researcher/tinkerer is doing something simple but hasn't messed with implementing the units BMI functions but is using an existing model that does, but he knows that the units match... maybe we shouldn't cater to that.

Reply:
What may make sense is to have a method within `UnitsHelper` that is always called on unit conversion failure... that can be the choke point for a configurable behavior and elimintates the "forgot to do it here" problem.

Reply:
Huh... So, after changing this to `throw`, all the tests still pass. I think I thought the legacy `Forcing` object would cause this case, but I guess it's not an issue because the two paths never cross... strange to me that this is the case though after I did the testing enhancements in #328 ... I suppose we can leave it like this, unless the tinkerer scenario above seems valid.